src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -369,29 +392,53 @@ def _purgeTree(self, ctx, baseDirPfn):
         :param ctx:  The gfal2 context object
         :return:     Bool: True if it managed to purge everything, False otherwise
         """
+        # NOTE: It deletes only directories and does not try to unlink any file.
+
         successList = []
 
+        # First test if baseDirPfn is actually a directory entry:
+        try:
+            entryStat = ctx.stat(baseDirPfn)
+        except gfal2.GError as gfalExc:
+            e = sys.exc_info()[1]
+            if e.code == errno.ENOENT:
+                self.logger.error("MISSING baseDir: %s", baseDirPfn)
+                successList.append(False)
```

Comment:
I would suggest to replace these two lines by a simple `return False`. Idem for the else block below.

Reply:
already the case